### PR TITLE
{debugger}[gompi/2020a,iimpi/2020a] Valgrind v3.15

### PR DESCRIPTION
**Pull request by maxim-masterov**
Includes patch to drop MPI-1 support. The patch is taken from [https://bugs.kde.org/show_bug.cgi?id=401416#c3](https://bugs.kde.org/show_bug.cgi?id=401416#c3).

### CODE REVIEW

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/62cf16372aa10c8ca93363380bbe02fd for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/7c322def9751fe6e500ea2798501168c for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
skalman.hpc2n.umu.se - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz, Python 2.7.17
See https://gist.github.com/b2a7246292fa35e97f1365b1aa316f4f for a full test report.

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-intel-2020a.eb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+toolchain = {'name': 'intel', 'version': '2020a'}
+toolchainopts = {'optarch': False}
+
+source_urls = [
+    'ftp://sourceware.org/pub/valgrind/',
+    'https://www.mirrorservice.org/sites/sourceware.org/pub/valgrind/',
+]
+sources = [SOURCELOWER_TAR_BZ2]
+patches = ['Valgrind-3.15.0-no-MPI-1.patch']
```
Comment (akesandgren) [inline easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-intel-2020a.eb]: This patch is technically speaking only required in the foss-2020a build., please remove it from the intel build.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
skalman.hpc2n.umu.se - Linux ubuntu 18.04, Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz, Python 2.7.17
See https://gist.github.com/c1a2727df476a023a50a32aa74e5036b for a full test report.

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-intel-2020a.eb (hunk 1 above)
Comment (maxim-masterov) [reply easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-intel-2020a.eb]: done

**Comment by lexming** [comment]
Re-triggering travis...

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
login2.cerberus.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/1dba824edc13c1344afe08ec40b105dd for a full test report.

**Comment by boegel** [comment]
close/re-open to get the right test result to merge this...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3401.kirlia.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/0505671db76e168b28f75659783bbf58 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2528.golett.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/92c8c9212fe4d9bbbef7c78b5c180ea9 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, x86_64, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/a1388d15647b1ddc7eb2315bac8aacdb for a full test report.

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+easyblock = 'ConfigureMake'
+
+name = 'Valgrind'
+version = '3.15.0'
+
+homepage = 'https://valgrind.org'
+description = "Valgrind: Debugging and profiling tools"
+
+toolchain = {'name': 'foss', 'version': '2020a'}
+toolchainopts = {'optarch': False}
```
Comment (boegel) [inline easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb:10]: @maxim-masterov Do you have a specific reason for disabling `optarch` here?

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+source_urls = [
+    'ftp://sourceware.org/pub/valgrind/',
+    'https://www.mirrorservice.org/sites/sourceware.org/pub/valgrind/',
+]
+sources = [SOURCELOWER_TAR_BZ2]
+patches = ['Valgrind-3.15.0-no-MPI-1.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb:17]: @maxim-masterov Please use `%(version)s` here:

```python
patches = ['Valgrind-%(version)s-no-MPI-1.patch']

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3128.skitty.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/b979ffacbf4ccde84fd8d0435816a790 for a full test report.

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb (hunk 2 above)
Comment (maxim-masterov) [reply easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb:10]: No particular reason, it was transferred from the previous config files. I see it down to Valgrind-3.13 version. Do you have any idea why?

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb (hunk 3 above)
Comment (maxim-masterov) [reply easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb]: Thanks, done!

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb:10]: See discussion in https://github.com/easybuilders/easybuild-easyconfigs/pull/3180, installation was failing back then, but maybe that's no longer an issue now with `optarch` enabled on recent processor architectures?

Comment (maxim-masterov) [reply easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb]: Okay. I've checked with foss and it works just fine with enabled optarch. However, a compilation with intel fails. So, I'm keeping it disabled for intel.

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+easyblock = 'ConfigureMake'
+
+name = 'Valgrind'
+version = '3.15.0'
+
+homepage = 'https://valgrind.org'
+description = "Valgrind: Debugging and profiling tools"
+
+toolchain = {'name': 'foss', 'version': '2020a'}
```
Comment (Micket) [inline easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb:9]: Surely, valgrind doesn't care about LAPACK, so, we should be able to drop this down to GCC level?
```suggestion
toolchain = {'name': 'GCC', 'version': '9.3.0'}
```

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-intel-2020a.eb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+easyblock = 'ConfigureMake'
+
+name = 'Valgrind'
+version = '3.15.0'
+
+homepage = 'https://valgrind.org'
+description = "Valgrind: Debugging and profiling tools"
+
+toolchain = {'name': 'intel', 'version': '2020a'}
```
Comment (Micket) [inline easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-intel-2020a.eb:9]:
```suggestion
toolchain = {'name': 'iccifort', 'version': '2020.1.217'}
```

File: easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb (hunk 4 above)
Comment (maxim-masterov) [reply easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb]: makes sense

Comment (maxim-masterov) [reply easybuild/easyconfigs/v/Valgrind/Valgrind-3.15.0-foss-2020a.eb]: pushed a new commit

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.7.1908, Intel Xeon Processor (Skylake), Python 2.7.5
See https://gist.github.com/6c65edfd78b4d328794cd4bfca93ae77 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 6 out of 6 (2 easyconfigs in this PR)
hebbe-c1 - Linux centos linux 7.7.1908, Intel Core Processor (Haswell, no TSX), Python 2.7.5
See https://gist.github.com/3a4ba7d679fd2a63c952bb04dc3c595c for a full test report.

**Pull request merged**